### PR TITLE
Created client interface for api events handler

### PR DESCRIPTION
**Pull request by kateknister**

### CODE REVIEW

File: api/handler.go [hunk 1]
```diff
@@ -163,6 +166,43 @@ func streamResults(results chan *events.Event, w http.ResponseWriter, r *http.Re
...
 }
 
+func requestResults(w http.ResponseWriter, r *http.Request) error {
+	oldUrlString := r.URL.RequestURI()
+	glog.Infof("oldUrlString is %v", oldUrlString)
+	requestElements := clientRegexp.FindStringSubmatch(oldUrlString)
+	if len(requestElements) != 3 {
+		return fmt.Errorf("Could not interpret client request: %v", requestElements)
+	}
+	newUrlString := "http://localhost:8080" + requestElements[1] + "events" + requestElements[2]
```
Comment (rjnagal) [inline api/handler.go]: Can you add comments on why we need to map it to localhost and hardcode port number here?

File: api/handler.go [hunk 2]
```diff
@@ -163,6 +166,43 @@ func streamResults(results chan *events.Event, w http.ResponseWriter, r *http.Re
...
+	glog.Infof("requestElements are %v", requestElements)
+	glog.Infof("newUrlString is %v", newUrlString)
+	req, err := http.NewRequest("GET", newUrlString, nil)
```
Comment (rjnagal) [inline api/handler.go]: Sorry, I don't have enough context as to what client API is supposed to do. From here, it looks like its rewriting the query to events endpoint and just print out the result. I haven't followed the events stuff in detail before. Can you describe it a bit here.

**Comment by kateknister** [comment]
Hey @rjnagal,  I moved the events client to the client library and got rid of the hardcoded url. Rather than printing, now the functions return a struct or a channel through which events may be retrieved. I put in an example main package as well but I can remove that, I wasn't sure if that was useful or not.

File: client/client.go [hunk 3]
```diff
@@ -39,10 +42,32 @@ func NewClient(url string) (*Client, error) {
...
 	return &Client{
-		baseUrl: fmt.Sprintf("%sapi/v1.2/", url),
+		baseUrl: fmt.Sprintf("%sapi/v1.3/", url),
 	}, nil
 }
 
+// Returns all past events that satisfy the request
+func (self *Client) EventStaticInfo(name string) (einfo []*events.Event, err error) {
+	u := self.eventsInfoUrl(name)
+	glog.Infof("got complete url %v", u)
```
Comment (rjnagal) [inline client/client.go]: Make the log message V(3).

File: api/handler.go [hunk 4]
```diff
@@ -57,6 +57,7 @@ func RegisterHandlers(mux httpMux.Mux, m manager.Manager) error {
 
 // Captures the API version, requestType [optional], and remaining request [optional].
 var apiRegexp = regexp.MustCompile("/api/([^/]+)/?([^/]+)?(.*)")
+var clientRegexp = regexp.MustCompile("(/api/[^/]+/?)client(.*)")
```
Comment (rjnagal) [inline api/handler.go]: Is this used anywhere?

**Comment by rjnagal** [comment]
@kateknister I think this looks good, except the one nit about unused var.

**Comment by rjnagal** [comment]
LGTM

**Comment by rjnagal** [comment]
Looks like one of the tests broke:

 github.com/google/cadvisor/client
client/client.go:50: undefined: events.Event
client/client.go:53: undefined: events.Event
# github.com/google/cadvisor/client
client/client.go:50: undefined: events.Event
client/client.go:53: undefined: events.Event
?   	github.com/google/cadvisor	[no test files]

**Pull request merged**